qiskit/visualization/matplotlib.py

```diff
@@ -304,7 +304,7 @@ def param_parse(self, v):
                 param_parts[i] = str(e)
 
             if param_parts[i].startswith('-'):
-                param_parts[i] = '$-$' + param_parts[i][1:]
+                param_parts[i] = '$-$' + '${}$'.format(param_parts[i][1:])
```

Comment:
This doesn't do anything. param_parts is already being formatted for math elsewhere. Instead of adding $-$ at the front, you need to replace it throughout param_parts.

Reply:
I understand it now. Thanks for pointing out. But, I am late to the party. Someone else made a PR to fix as you suggested

Reply:
If you want to finish this, go ahead.

Reply:
Thank you for giving me a go. I'll proceed where I left